CODE RUB: AIFiles Exception Classes Update v2.10.1

Lezune opened the pull request:
Updated All Exceptions associated to the AIFiles directory to adhere to Anemic Data Models as well as updated the service layer to include the exception messaging.

glhays commented on Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/AIFileDependencyException.cs:
```diff
@@ -8,11 +8,6 @@ namespace Standard.AI.OpenAI.Models.Services.Foundations.AIFiles.Exceptions
-                message: "AI file dependency error occurred, contact support.",
-                    innerException: innerException)
-        { }
 
```
Remove extra line

Lezune replied:
This has been completed. Sorry for the wait.

glhays commented on Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/AIFileDependencyValidationException.cs:
```diff
@@ -8,11 +8,7 @@ namespace Standard.AI.OpenAI.Models.Services.Foundations.AIFiles.Exceptions
-                message: "AI file dependency validation error occurred, contact support.",
-                    innerException: innerException)
-        { }
+
```
Extra line remove

glhays commented on Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/ExcessiveCallAIFileException.cs:
```diff
@@ -9,11 +9,6 @@ namespace Standard.AI.OpenAI.Models.Services.Foundations.AIFiles.Exceptions
-                message: "Excessive call error occurred, limit your calls.",
-                    innerException: innerException)
-        { }
 
```
Extra line remove

Lezune replied:
This has been completed. Sorry for the wait.

glhays commented on Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/FailedAIFileServiceException.cs:
```diff
@@ -9,11 +9,6 @@ namespace Standard.AI.OpenAI.Models.Services.Foundations.AIFiles.Exceptions
-                message: "Failed AI file service error occurred, contact support.",
-                    innerException: innerException)
-        { }
 
```
Same

Lezune replied:
This has been completed. Sorry for the wait.

glhays commented on Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/FailedServerAIFileException.cs:
```diff
@@ -9,11 +9,6 @@ namespace Standard.AI.OpenAI.Models.Services.Foundations.AIFiles.Exceptions
-                message: "Failed AI file server error occurred, contact support.",
-                    innerException: innerException)
-        { }
 
```
Same

glhays commented on Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/InvalidAIFileException.cs:
```diff
@@ -9,16 +9,10 @@ namespace Standard.AI.OpenAI.Models.Services.Foundations.AIFiles.Exceptions
-                message: "Invalid AI file error occurred, fix errors and try again.",
-                    innerException: innerException)
-        { }
 
```
Extra line remove

glhays commented on Standard.AI.OpenAI/Models/Services/Foundations/AIModels/Exceptions/AIModelServiceException.cs:
```diff
@@ -8,11 +8,6 @@ namespace Standard.AI.OpenAI.Models.Services.Foundations.AIModels.Exceptions
-                message: "AI Model service error occurred, contact support.",
-                    innerException: innerException)
-        { }
 
```
Extra line remove

glhays commented on Standard.AI.OpenAI/Models/Services/Foundations/AIModels/Exceptions/AIModelValidationException.cs:
```diff
@@ -8,11 +8,6 @@ namespace Standard.AI.OpenAI.Models.Services.Foundations.AIModels.Exceptions
-                message: "AI Model validation error occurred, fix errors and try again.",
-                    innerException: innerException)
-        { }
 
```
same

glhays commented on Standard.AI.OpenAI.Tests.Unit/Services/Foundations/AIFiles/AIFileServiceTests.Validations.RemoveById.cs:
```diff
@@ -24,7 +24,7 @@ public partial class AIFileServiceTests
 
             var invalidFileException =
-                new InvalidAIFileException();
+                new InvalidAIFileException(message: "Invalid AI file error occurred, fix errors and try again.");
```
```suggestion
                new InvalidAIFileException(
                    message: "Invalid AI file error occurred, fix errors and try again.");
```

glhays commented on Standard.AI.OpenAI.Tests.Unit/Services/Foundations/AIFiles/AIFileServiceTests.Validations.Upload.cs:
```diff
@@ -52,7 +52,7 @@ private async Task ShouldThrowValidationExceptionOnUploadIfAIFileRequestIsNull()
             var invalidAIFileException =
-                new InvalidAIFileException();
+                new InvalidAIFileException(message: "Invalid AI file error occurred, fix errors and try again.");
 
```
```suggestion
               new InvalidAIFileException(
                message: "Invalid AI file error occurred, fix errors and try again.");
```

glhays commented on Standard.AI.OpenAI.Tests.Unit/Services/Foundations/AIFiles/AIFileServiceTests.Validations.Upload.cs:
```diff
@@ -99,7 +99,7 @@ private async Task ShouldThrowValidationExceptionOnUploadIfAIFileRequestIsInvali
 
             var invalidAIFileException =
-                new InvalidAIFileException();
+                new InvalidAIFileException(message: "Invalid AI file error occurred, fix errors and try again.");
```
```suggestion
                new InvalidAIFileException(
                    message: "Invalid AI file error occurred, fix errors and try again.");
```

Lezune commented:
Hello @glhays Greg,

Thank you so much for the information as it is much appreciated. I will follow your advice and will study up on the best practices and guidelines to ensure I do not add unnecessary stress or work to the team. I will make sure I update the titles of my other pull requests after I reed the guidelines. Thanks again and I look forward to working with you.

glhays commented on Standard.AI.OpenAI.Tests.Unit/Services/Orchestrations/AIFiles/AIFileOrchestrationServiceTests.cs:
```diff
@@ -46,8 +46,8 @@ public static TheoryData DependencyValidationExceptions()
                 new LocalFileDependencyValidationException(someInnerException),
-                new AIFileValidationException(someInnerException),
-                new AIFileDependencyValidationException(someInnerException)
+                new AIFileValidationException(message: "AI file validation error occurred, fix errors and try again.", someInnerException),
+                new AIFileDependencyValidationException(message: "AI file dependency validation error occurred, contact support.", someInnerException)
```
These lines exceed the 120 character line length limit, when they are broken then they need to be seperated by an extra line.

```suggestion

                new AIFileValidationException(
                    message: "AI file validation error occurred, fix errors and try again.",
                    innerException: someInnerException),
                
                new AIFileDependencyValidationException(
                    message: "AI file dependency validation error occurred, contact support.",
                    innerException: someInnerException)
```

glhays commented on Standard.AI.OpenAI.Tests.Unit/Services/Orchestrations/AIFiles/AIFileOrchestrationServiceTests.cs:
```diff
@@ -59,8 +59,8 @@ public static TheoryData DependencyExceptions()
                 new LocalFileServiceException(someInnerException),
-                new AIFileDependencyException(someInnerException),
-                new AIFileServiceException(someInnerException),
+                new AIFileDependencyException(message : "AI file dependency error occurred, contact support.", someInnerException),
```
Same here with these two lines

glhays commented on Standard.AI.OpenAI.Tests.Unit/Services/Orchestrations/AIFiles/AIFileOrchestrationServiceTests.cs:
```diff
@@ -70,8 +70,8 @@ public static TheoryData AIFileServiceDependencyExceptions()
             {
-                new AIFileDependencyException(someInnerException),
-                new AIFileServiceException(someInnerException),
+                new AIFileDependencyException(message : "AI file dependency error occurred, contact support.", someInnerException),
```
Same Here

glhays reviewed (commented):
@Lezune 
In your Service Classes Exceptions and Validations, these need refactored use the Service classes in ADotNet as an example. [ADotNet](https://github.com/The-Standard-Organization/ADotNet/blob/main/ADotNet/Services/Foundations/Files/FileService.Exceptions.cs)

glhays commented:
@Lezune are you on the discord channel?

Lezune commented:
@glhays I am on te discord channel. I changed my username to Lezune to make it easier to spot me. Also, thank you for the feedback.

glhays commented:
If my comments are fixed, please mark the comment resolved in the git IDE.

On Thu, May 30, 2024, 9:13 AM Lezune ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In
> Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/FailedAIFileServiceException.cs
> <https://github.com/hassanhabib/Standard.AI.OpenAI/pull/436#discussion_r1621034924>
> :
>
> > @@ -9,11 +9,6 @@ namespace Standard.AI.OpenAI.Models.Services.Foundations.AIFiles.Exceptions
>  {
>      public class FailedAIFileServiceException : Xeption
>      {
> -        public FailedAIFileServiceException(Exception innerException)
> -            : base(
> -                message: "Failed AI file service error occurred, contact support.",
> -                    innerException: innerException)
> -        { }
>
>
> This has been completed. Sorry for the wait.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hassanhabib/Standard.AI.OpenAI/pull/436#discussion_r1621034924>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKD4KOYQZ2ILD26OCCVAWJTZE5F3NAVCNFSM6AAAAABINXEGYOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDAOBYG43DSNBUHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Lezune commented:
Thank you for all of the help. All conversations have been marked as resolved.

Lezune commented:
Updated and refactored code to reduce redundancy by adding methods in the AIFilesService class and calling the methods in the AIFilesService.Exception and AIFilesService.Validation classes as suggested by @glhays. 

Please ignore AIModel classes as they should be processes in the following Pull Requests:

#507 
#508 
#509 
#510 
#511 
#512 
#513 
#514 
#515 
#516 
#517

glhays commented on Standard.AI.OpenAI/Models/Services/Foundations/AIModels/Exceptions/NotFoundAIModelException.cs:
```diff
@@ -9,11 +9,6 @@ namespace Standard.AI.OpenAI.Models.Services.Foundations.AIModels.Exceptions
-                message: "AI Model not found.",
-                    innerException: innerException)
-        { }
 
```
Remove extra line

Lezune replied:
This has been resolved.

glhays commented on Standard.AI.OpenAI.Tests.Unit/Services/Foundations/AIFiles/AIFileServiceTests.Exceptions.RemoveById.cs:
```diff
@@ -25,13 +26,11 @@ private async Task ShouldThrowDependencyExceptionOnRemoveByIdIfUrlNotFoundAsync(
             var invalidConfigurationFileException =
-                new InvalidConfigurationAIFileException(
-                    message: "Invalid AI file configuration error occurred, contact support.",
+                createInvalidConfigurationAIFileException(
```
Remove all these in the unit test methods, as explained in Discord.
This comment refers to all, and will not repeat the comment for each instance there is a (create)

Lezune replied:
The Test unit files in the AIFiles directory were reverted to their original state. Had to update several lines as well as add another create method to resolve an error cause by a change in the Constructor of one of the updated Exceptions.

glhays commented:
@Lezune I suggest that you do a similar merge with this PR, your not in an appropriate branch like
`user/lezune/coderub-aifiles-upgarde-v2101` but your going from from your forks main to main.
Don't worry about the comments you can always open the closed PR to review if needed.

Lezune commented:
Thank you for the advice, @glhays. I will follow your suggestion and merge this into a new branch later today.

Lezune commented:
Closing Pull Request. Resolved in PR #558.
